pkg/fileutil/io.go

```diff
@@ -131,3 +136,34 @@ func FileExists(p string) (bool, error) {
 	}
 	return false, fmt.Errorf("%s: %w", p, ErrNotFile)
 }
+
+func VerifyAbsPath(absPath, basePath string) error {
+	// check we have a valid abs path
+	if !path.IsAbs(absPath) || path.Clean(absPath) != absPath {
+		return ErrBadPath
+	}
+	// point to storage namespace
+	if !strings.HasPrefix(absPath, basePath) {
+		return ErrInvalidPath
+	}
+	return nil
+}
+
+func VerifyRelPath(relPath, base string) error {
+	abs := base + string(os.PathSeparator) + relPath
+	return VerifyAbsPath(abs, base)
+}
+
+func VerifyRegularFile(name, base string) error {
```

Comment:
This function is good but has a confusing name: it doesn't verify that `name` is regular: there are many types of strange files other than symlinks that are not regular, including directories, devices, named pipes, ...
```suggestion
func VerifySafeFilename(name, base string) error {
```

Also please document what `base` means here, for instance see my confusion on the very next line.  And consider switching their order perhaps?